### PR TITLE
MueLu: classical with Tpetra

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/muelu 

## Motivation
Fix a bug in Tpetra classical.

### CODE REVIEW

**Review by csiefer2 — DISMISSED** [review]

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalPFactory_def.hpp [hunk 1]
```diff
@@ -697,7 +707,7 @@ Coarsen_ClassicalModified(const Matrix & A,const RCP<const Matrix> & Aghost, con
                 GO m_g    = Aghost->getColMap()->getGlobalElement(A_indices_k[m0]);
                 LO mghost = A_indices_k[m0];//Aghost LCID
                 LO m = Aghostcol_to_Acol[mghost]; //A's LID (could be LO_INVALID)
-                if(m_g != k_g && m != LO_INVALID && Acol_to_Pcol[A_indices_k[m0]] >=  (LO)P_rowptr[i]) {
+                if(m_g != k_g && m != LO_INVALID && Acol_to_Pcol[m] >=  (LO)P_rowptr[i]) {
```
Comment (cgcgcg) [inline packages/muelu/src/Transfers/Classical/MueLu_ClassicalPFactory_def.hpp:710]: Is that ok?

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalPFactory_def.hpp [hunk 2]
```diff
@@ -711,7 +721,7 @@ Coarsen_ClassicalModified(const Matrix & A,const RCP<const Matrix> & Aghost, con
                   LO jghost = A_indices_k[j0];//Aghost LCID
                   LO j = Aghostcol_to_Acol[jghost]; //A's LID (could be LO_INVALID)
                   // NOTE: Row k should be in fis_star, so I should have to check for diagonals here
-                  if(Acol_to_Pcol[j] >=  (LO)P_rowptr[i]) {
+                  if((j != LO_INVALID) && (Acol_to_Pcol[j] >=  (LO)P_rowptr[i])) {
```
Comment (cgcgcg) [inline packages/muelu/src/Transfers/Classical/MueLu_ClassicalPFactory_def.hpp:724]: How about this?

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalPFactory_def.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/src/Transfers/Classical/MueLu_ClassicalPFactory_def.hpp:710]: I believe that is correct.

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalPFactory_def.hpp (hunk 2 above)
Comment (csiefer2) [reply packages/muelu/src/Transfers/Classical/MueLu_ClassicalPFactory_def.hpp:724]: Good catch!

**Review by csiefer2 — APPROVED** [review]

**Pull request merged**